_posts/2023-02-22-user-defined-operators.md

```diff
@@ -0,0 +1,484 @@
+#### Built-in operators
+Solidity, like the vast majority of programming languages,
+[offers a set of operators](https://docs.soliditylang.org/en/latest/types.html#operators) as syntactic sugar
+over common mathematical functions.
+The difference between operators and functions is purely syntactical, and the decision
+to choose one over the other is to a large extent arbitrary.
+What is included in that set, however, varies widely from language to language.
+
+Here is the full list for Solidity:
+
+| Category             | Operators                                                                |
+|----------------------|--------------------------------------------------------------------------|
+| Bitwise              | `&`, `\|`, `^`, `<<`, `>>`, `~`                                          |
+| Arithmetic           | `+`, `-`, `*`, `/`, `%`, `**`                                            |
+| Comparison           | `==`, `!=`, `<`, `<=`, `>`, `>=`                                         |
+| Boolean              | `!`, `&&`, `\|\|`                                                        |
+| Increment/decrement  | `++`, `--`                                                               |
+| Simple assignment    | `=`                                                                      |
+| Composite assignment | `+=`, `-=`, `^=`, `*=`, `/=`, `%=`, `&=`, `\|=`, `^=`, `<<=`, `>>=`      |
+| Miscellaneous        | `delete`, `new`, `.`, `()`, `[]`, `?:`                                   |
+
+Most of them require no explanation.
+In some cases, however, you may be surprised by what counts as an operator.
+The common infix operator, like `a + b`, that may be on your mind when you hear "operator" is
+not the only flavor operators come in.
+The last row of the table has items that to some users may not seem operator-like at all.
+`delete` and `new`  are in fact operators, and follow precedence and associativity rules despite not being symbols.
+Member access via `.`, subscripting arrays with `[]`, or making a function call by adding parentheses
+to a function can be thought of as applications of operators as well.
+Even putting something in parentheses creates a valid operator expression.
+
+It is important to note that in many languages assignment is a statement that is handled specially.
+In Solidity, like in the C-like languages it took inspiration from, assignment can be freely used in
+expressions on par with other operators.
+
+In most cases built-in operators are only defined for [value types](https://docs.soliditylang.org/en/latest/types.html#value-types).
+Notable exceptions are the simple assignment and the "miscellaneous" group.
+For example, you can put almost anything in parentheses or branches of the ternary operator, as a part
+of a bigger expression, even if its type cannot be named or even assigned to anything.
+
+#### `using for`
+Since operators are a purely syntactic element, you can always use ordinary functions as a substitute.
```

Comment:
> I would have thought the ordinary functions are free functions without "using for". 

Well, it's any function. I mean, you can just call a function anywhere where you'd use an operator and often you have many kinds of functions to choose from. Changed to just `a function`.

> Maybe we should not compare the existing "using for" with operators but rather explain it in isolation?

I'd prefer to keep it that way, because really the main reason I added it is because of operators and I wanted to underscore the elements that make them similar in some ways.